src/server/pfs/server/http.go

```diff
@@ -107,27 +114,44 @@ type loginRequestPayload struct {
 }
 
 func (s *HTTPServer) authLoginHandler(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
-	var data loginRequestPayload
-	decoder := json.NewDecoder(r.Body)
-	err := decoder.Decode(&data)
-	if err != nil {
-		// Return 500
-		http.Error(w, fmt.Sprintf("malformed JSON sent in payload: %v", err), http.StatusInternalServerError)
-		return
-	}
-	if data.Token == "" {
+	token := r.FormValue("Token")
+	if token == "" {
 		// Return 500
 		http.Error(w, "empty token provided", http.StatusInternalServerError)
 		return
 	}
-	w.Header().Add("Set-Cookie", fmt.Sprintf("%v=%v", auth.ContextTokenKey,
-		data.Token))
+	w.Header().Add("Set-Cookie", fmt.Sprintf("%v=%v;path=/", auth.ContextTokenKey,
+		token))
 	w.Header().Add("Access-Control-Allow-Origin", "*")
 	w.WriteHeader(http.StatusOK)
 }
 
 func (s *HTTPServer) authLogoutHandler(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
-	w.Header().Add("Set-Cookie", fmt.Sprintf("%v=", auth.ContextTokenKey))
+	w.Header().Add("Set-Cookie", fmt.Sprintf("%v=;path=/", auth.ContextTokenKey))
 	w.Header().Add("Access-Control-Allow-Origin", "*")
 	w.WriteHeader(http.StatusOK)
 }
+
+type notFoundRouter struct {
+}
+
+func (s notFoundRouter) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	w.Header().Add("Content-Type", "text/html; charset=utf-8")
+	w.WriteHeader(http.StatusNotFound)
```

Comment:
One more nit: could you use `http.Error(w, "route not found", http.StatusNotFound)` here?

Reply:
Yup, good call